commands/Idle.java

```diff
@@ -1,46 +1,62 @@
 package org.usfirst.frc4904.standard.commands;
 
+import java.util.Set;
 
-import java.util.StringJoiner;
-import edu.wpi.first.wpilibj.command.Command;
-import edu.wpi.first.wpilibj.command.Subsystem;
+import org.usfirst.frc4904.standard.LogKitten;
 
-public class Idle extends Command {
-	public Idle() {
-		super("Idle[No Subsystem]");
-		setInterruptible(true);
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.Subsystem;
+
+/**
+ * Requires subsystems so no other code can interact with them.
+ */
+public class Idle extends CommandBase {
+	boolean verbose;
+	Set<Subsystem> requirements;
+
+	public Idle(String name, boolean verbose, Subsystem... subsystems) {
+		super();
+		setName(name);
+		this.verbose = verbose;
+		requirements = Set.of(subsystems);
+	}
+
+	public Idle(boolean verbose, Subsystem... subsystems) {
+		this("Idle", verbose, subsystems);
 	}
 
 	public Idle(Subsystem... subsystems) {
-		super("Idle[" + Idle.joinSubsystemNames(subsystems) + "]");
-		setInterruptible(true);
-		for (Subsystem subsystem : subsystems) {
-			requires(subsystem);
-		}
+		this("Idle", false, subsystems);
 	}
 
-	protected static String joinSubsystemNames(Subsystem[] subsystems) {
-		StringJoiner subsystemNames = new StringJoiner("|");
-		for (Subsystem subsystem : subsystems) {
-			subsystemNames.add(subsystem.getName());
+	public void initialize() {
+		if (verbose) {
+			LogKitten.v("Idle " + getName() + " initialized.");
 		}
-		return subsystemNames.toString();
 	}
 
-	@Override
-	protected void initialize() {}
-
-	@Override
-	protected void execute() {}
+	public void execute() {
+		if (verbose) {
+			LogKitten.v("Idle " + getName() + " executed.");
+		}
+	}
 
-	@Override
-	protected boolean isFinished() {
+	public boolean isFinished() {
+		if (verbose) {
+			LogKitten.v("Idle " + getName() + " isFinished?");
+		}
 		return false;
 	}
 
-	@Override
-	protected void end() {}
+	protected void end() {
```

Comment:
`end` takes in a `boolean interrupted`

Reply:
also call the superclass's implementation of `end`

Reply:
not sure how to fix